adserver/reports.py

```diff
@@ -193,7 +193,15 @@ def generate(self):
 
         results = {}
 
-        for impression in queryset:
+        for impression in queryset.only(
+            "advertisement__flight",
+            "publisher__revenue_share_percentage",
+            "decisions",
+            "offers",
+            "views",
+            "clicks",
+            self.index,
```

Comment:
This needs a bit more logic to work properly -- but I think this change is good with a bit of tuning.

Reply:
Ya, I think we might need to translate dot notation to dunder notation or something.